debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -775,7 +816,7 @@ public OffsetStorageReader offsetStorageReader() {
                     // Start the connector with the given properties and get the task configurations ...
                     connector.start(workerConfig.originalsStrings());
                     connectorCallback.ifPresent(DebeziumEngine.ConnectorCallback::connectorStarted);
-                    List<Map<String, String>> taskConfigs = connector.taskConfigs(1);
+                    List<Map<String, String>> taskConfigs = connector.taskConfigs(maxTasks);
```

Comment:
After discussing with @Naros came to the conclusion that this is potentially problematic. When running server with multiple tasks each embedded engine will perform its own call to ``connector.taskConfigs(maxTasks)``. However, there is no guarantee that the result of these calls won't be different.  Current approach works just fort connectors with static task configurations and for example Debezium's MongoDb connector would be error prone as it retrieves the current replicaset configuration (which could change  between server starting multiple engines), furthermore the connector requests task reconfiguration in case of change to replica sets. 

A partial solution (not dealing with the reconfiguration requests) would be to push the for loop from DebeziumServer down to EmbeddedEngine directly, having Engine run multiple tasks, each in its own thread.

Reply:
@jpechane  WDY think? 

@vrajat  Are you still interested in working on this? Depending on how the change goes it might be above my currently feasible time allocation.

Reply:
What is a good medium to discuss and write a design doc? I agree that if `taskConfigs` doesn't return a stable set, then this approach will not work. Is it practical to add a new API that should return stable results ? A connector can choose to implement it or not.

One advantage of multiple engines is that the complete pipeline is parallel including the transformers and sink connectors.

Reply:
IMHO it makes sense to push the multithreading down to Engine. The Debezium Engine API will be extended with an option to pass `ThreadFactory`.

When working it makes sense to look at reconfiguration.

Reply:
If I understand you correctly, the first part of `EmbeddedEngine::run()` will get the configuration and setup the tasks. Then there will be potentially multiple threads executing the [while loop](https://github.com/debezium/debezium/blob/main/debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java#L833) using the task configs assigned to them?

Reply:
@vrajat  Correct. Apology for the late answer, I was out sick.